src/scripting/daedalus/DaedalusVMForGameWorld.cpp

```diff
@@ -196,11 +440,35 @@ namespace REGoth
       mStack.pushInstance((SymbolIndex)symbolIndex);
     }
 
+    void DaedalusVMForGameWorld::external_HLP_IsValidNpc()
+    {
+      HCharacter character = popCharacterInstance();
+
+      mStack.pushInt(!!character ? 1 : 0);
```

Comment:
Sorry to pick on this again but why not just do `mStack.pushInt(character ? 1 : 0);`? `bs::GameObjectHandle` allows *direct* conversion to bool.

Reply:
I think the direct conversion to bool is fine in if statements, but it just looks weird in these cases. I'll refactor it to an explicit `if`, if you don't mind.

Reply:
The explicit `if` looks even more strange :D. Like
```cpp
if(true)
{
  return true;
}
else
{
  return false;
}
```

With stl smart pointers you often do exactly this:
```
if(pointer) ...

pointer ? ... : ...
```
It's just the short way of writing `pointer == nullptr` and I think the intent is pretty much clear.

But if you don't want to do it that way, better keep the double negation because that somewhat even looks cleaner than the explicit way.

Reply:
I agree your first example looks weird. But remember that we don't check a constant `true`, which is some part of the weirdness. Also, we don't return `true` nor `false` but we call a method to push something on the stack with the intefers `1` or `0`. I assume that makes it look a lot less weird.